src/booleans.js

```diff
@@ -1,62 +1,68 @@
 function negate(a) {
-  // your code here
+  return !a;
+  //
 };
 
 function both(a, b) {
-  // your code here
+  return a && b;
 };
 
 function either(a, b) {
-  // your code here
+  return a || b;
 };
 
 function none(a, b) {
-  // your code here
+  return !a && !b;
 };
 
 function one(a, b) {
-  // your code here
+  return (a&&!b) || (b&&!a);
 };
 
 function truthiness(a) {
-  // your code here
+  return !!a;
 };
 
 function isEqual(a, b) {
-  // your code here
+  return !!(a==b) ;
 };
 
 function isGreaterThan(a, b) {
-  // your code here
+  return (a>b);
 };
 
 function isLessThanOrEqualTo(a, b) {
-  // your code here
+  return (a<=b);
 };
 
 function isOdd(a) {
-  // your code here
+  return !((a%2)==0);
 };
 
 function isEven(a) {
-  // your code here
+  return ((a%2)==0);
 };
 
 function isSquare(a) {
-  // your code here
+  return a>=0&&Math.sqrt(a)%1===0;
 };
 
 function startsWith(char, string) {
-  // your code here
+  return string.startsWith(char);
 };
 
 function containsVowels(string) {
-  // your code here
-};
+  string.toLowerCase();
+  let m = string.match(/[aeiou]/gi);
+  return m === null ? false : true;
```

Comment:
Nice use of a ternary!

Reply:
You could simplify this as we know null is falsy. You could just return `m === null`

Reply:
We should always try to name our variable so they describe what they represent. So in this case a better variable name would be length or arrayLength, rather than using m.

Reply:
It's good you used `let` here, but because you are never reassigning the value of `m`, you can use `const m` instead.